directives/animate/animate.js

```diff
@@ -290,52 +289,37 @@ class AnimationState {
 
 }
 
-export class Hide extends AsyncDirective {
-	state;
-	constructor(propertyPart) {
-		super(propertyPart);
-		if (!(propertyPart.type === PropertyPart) || propertyPart.committer.name !== 'animate') {
-			throw new Error('animation directives must be used with "animate" property');
-		}
-	}
-	// Do SSR-compatible rendering (arguments are passed from call site)
-	render(part) {
-		// Previous state available on class field
-		if (this.state === undefined) {
-			this.state = new AnimationState(part);
-		}
+class Hide extends AsyncDirective {
+	render() {
 		return noChange;
 	}
-	update(part, opts) {
-		/* Any imperative updates to DOM/parts would go here */
+	update(part, [opts]) {
 		opts = opts || {};
-		this.state.hide(opts);
-		return this.render(part);
+		let state = stateMap.get(part.element);
```

Comment:
Right, so instead of the class property we now use the stateMap to store the state.